src/Board.cpp

```diff
@@ -0,0 +1,211 @@
+        targetPiece.reset(nullptr);
+    }
+
+    // Move the piece from the start position to the end position.
+    std::swap(m_chessMatrix[startPos.first][startPos.second], m_chessMatrix[endPos.first][endPos.second]);
+    m_currentPlayer = !m_currentPlayer;
+}
+
+/**
+ * Checks if the given color is in check.
+ * @param color - The color to check if in check.
+ * @return - True if the given color is in check, false otherwise.
+ */
+bool Board::isCheck(const bool color) const {
+
+    // Find the position of the king of the given color.
+    std::pair<int, int> kingPosition;
+
+    for (int row = 0; row < m_numRows; ++row) {
+        for (int col = 0; col < m_numCols; ++col) {
+            if (m_chessMatrix[row][col] &&
+                typeid(*m_chessMatrix[row][col]) == typeid(King) &&
+                m_chessMatrix[row][col]->getColor() == color) {
+                kingPosition = std::make_pair(row, col);
+                break;
+            }
+        }
+    }
+
+    // Check if any piece of the opposite color can capture the king.
+    for (int row = 0; row < 8; ++row) {
+        for (int col = 0; col < 8; ++col) {
+            if (m_chessMatrix[row][col] != nullptr && m_chessMatrix[row][col]->getColor() != color) {
+                std::pair piecePos = { row, col };
+                if (m_chessMatrix[row][col]->isValidMove({ piecePos, kingPosition }, m_chessMatrix) && isValidPath(piecePos, kingPosition)) {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Checks if the path between two given positions is valid.
+ * @param currPos - The current position.
+ * @param newPos - The new position.
+ * @return - True if the path is valid, false otherwise.
+ */
+bool Board::isValidPath(const std::pair<int, int>& currPos, const std::pair<int, int>& newPos) const {
+    int deltaRow = newPos.first - currPos.first;
+    int deltaCol = newPos.second - currPos.second;
+
+    // Calculate the direction of the move.
+    int rowDir = (deltaRow == 0) ? 0 : deltaRow / abs(deltaRow);
+    int colDir = (deltaCol == 0) ? 0 : deltaCol / abs(deltaCol);
+
+    // Check if the path is clear.
+    int checkRow = currPos.first + rowDir;
+    int checkCol = currPos.second + colDir;
+
+    while (checkRow != newPos.first || checkCol != newPos.second) {
+        if (m_chessMatrix[checkRow][checkCol] != nullptr) {
+            return false;
+        }
+        checkRow += rowDir;
+        checkCol += colDir;
+    }
+
+    return true;
+}
+
+/**
+ *  @brief Constructor for the Board class.
+ *  Initializes the chess matrix to contain null pointers for each square on the board.
+ *  @return void
+ */
+Board::Board() {
+    for (int row = 0; row < m_numRows; ++row) {
+        std::vector<std::unique_ptr<Piece>> rowVec;
+        for (int col = 0; col < m_numCols; ++col) {
+            rowVec.emplace_back(nullptr);
+        }
+        m_chessMatrix.emplace_back(std::move(rowVec));
+    }
+}
```

Comment:
Consider reordering the functions in a logical manner, such as starting with the public interface functions followed by helper functions. This helps readers understand the public interface of the class first.

Reply:
Constructor is at the bottom, because it's in the private section.